synapse/handlers/room.py

```diff
@@ -1139,13 +1139,21 @@ async def create_event(
         depth += 1
         state_map[(EventTypes.Member, creator.user.to_string())] = member_event_id
 
+        # we need the state group of the membership event as it is the current state group
+        event_to_state = (
+            await self._storage_controllers.state.get_state_group_for_events(
+                [member_event_id]
+            )
+        )
+        current_state_group = event_to_state[member_event_id]
+
         events_to_send = []
         # We treat the power levels override specially as this needs to be one
         # of the first events that get sent into a room.
         pl_content = initial_state.pop((EventTypes.PowerLevels, ""), None)
         if pl_content is not None:
             power_event, power_context = await create_event(
-                EventTypes.PowerLevels, pl_content, False
+                EventTypes.PowerLevels, pl_content, True
```

Comment:
this bool seems to be `for_batch`, which doesn't have a docstring.
Would you mind adding a docstring on it? I would find it especially useful to know why it's needed and what the semantics are — from the name I'm guessing the usage is just `True when you are creating an event to send in a batch`, but the other parts are more interesting

Reply:
I've added a docstring, let me know if it makes more sense.

Reply:
makes sense, but maybe would benefit from an elaboration about what the difference is